### PR TITLE
Revert LTO change for all variants, except for the NRF52/test

### PR DESCRIPTION
**Pull request by szszszsz**
Always use `LTO=fat`, unless NRF52/test is built - then use `LTO=thin`

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
[Custom profiles](https://doc.rust-lang.org/cargo/reference/profiles.html#custom-profiles) would probably be a better option here. I think the current approach would cause a complete recompilation of the entire project each time `Cargo.toml` is touched by the `sed` call in the makefile.

**Comment by szszszsz** [comment]
"Test" feature detection is a bit flaky here - ideas to improve?

File: Cargo.toml [hunk 1]
```diff
@@ -38,10 +38,17 @@ trussed-usbip = { git = "https://github.com/Nitrokey/pc-usbip-runner", tag = "v0
 
 [profile.release]
 codegen-units = 1
-lto = "thin"
+lto = "fat"
 opt-level = "z"
 incremental = false
 debug = true
 
+[profile.release-thin]
```
Comment (sosthene-nitrokey) [inline Cargo.toml:46]: I think `thin-lto` should be the name of the profile.

**Review by sosthene-nitrokey — APPROVED** [review]

**Comment by sosthene-nitrokey** [comment]
It's ugly but it's a "temporary" hack. Hopefully we won't need it for long, therefore I don't mind the bash uglyness.

**Comment by szszszsz** [comment]
Python version is better

File: runners/embedded/Makefile [hunk 2]
```diff
@@ -130,9 +130,10 @@ build: build-banner $(SRCS) check-var-BOARD check-var-BUILD_PROFILE check-var-SO
 	echo 'build_profile = "$(BUILD_PROFILE)"' >> cfg.toml
 	echo 'board = "$(BOARD)"' >> cfg.toml
 
-	cargo build --release --target $(TARGET) \
+	# NRF52/test -> "release-thin", use "release" otherwise
+	cargo build --target $(TARGET) \
 		--features $(COMMA_FEATURES) \
-		--quiet
+		--quiet --profile $(shell python -c "p = 'release-thin' if '$(BOARD)' == 'nk3am' and 'test' in '$(COMMA_FEATURES)'.split(',')  else 'release'; print(p); " )
```
Comment (robin-nitrokey) [inline runners/embedded/Makefile:136]: Can you please put this in a variable to make the `cargo` invocation easier to read?

File: Cargo.toml (hunk 1 above)
Comment (szszszsz) [reply Cargo.toml]: Changed to `release-thin-lto`

File: runners/embedded/Makefile (hunk 2 above)
Comment (szszszsz) [reply runners/embedded/Makefile]: Done

**Comment by szszszsz** [comment]
CI reports green

**Comment by szszszsz** [comment]
Requesting a brief final look before merging.

**Pull request merged**